doc/release/upcoming_changes/15997.improvement.rst

```diff
@@ -0,0 +1,3 @@
+Improve `array_repr`
+--------------------
+Now `array_repr` returns nicely formatted repr for arrays with objects with multi-line reprs.
```

Comment:
It's perhaps worth explaining this in simpler terms - this affects `repr(arr)`, not just `np.array_repr(arr)` - so any user of an interactive prompt will see the improvement.